freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/FreeIpaDeletionService.java

```diff
@@ -42,9 +61,39 @@ public void delete(String environmentCrn, String accountId) {
     }
 
     private void unscheduleAndTriggerTerminate(Stack stack) {
+        flowCancelService.cancelTooOldTerminationFlowForResource(stack.getId(), stack.getName());
         freeipaJobService.unschedule(stack);
+        if (!stack.isDeleteCompleted()) {
+            handleIfStackIsNotTerminated(stack);
+        } else {
+            LOGGER.debug("Stack is already deleted.");
+        }
+    }
+
+    private void handleIfStackIsNotTerminated(Stack stack) {
+        LOGGER.info("stack {} in environment {} is not deleted.", stack.getName(), stack.getEnvironmentCrn());
+        Optional<FlowLog> optionalFlowLog = findLatestTerminationFlowLogWithInitState(stack);
+        if (optionalFlowLog.isPresent()) {
+            FlowLog flowLog = optionalFlowLog.get();
+            LOGGER.debug("Found termination flowlog with id [{}] and payload [{}]", flowLog.getFlowId(), flowLog.getPayload());
+        } else {
+            fireTerminationEvent(stack);
+        }
+    }
+
+    private void fireTerminationEvent(Stack stack) {
+        LOGGER.warn("Couldn't find FlowLog with 'INIT_STATE'. Triggering termination");
```

Comment:
it is not informative. In the logs, we cannot find easily what flow does not have `INIT_STATE`

Reply:
modified to debug, it's only to check the flow of the code if there is an issue